src/main/java/md/maib/service/impl/CustomerServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package md.maib.service.impl;
+
+import md.maib.entity.Customer;
+import md.maib.repository.CustomerRepository;
+import md.maib.service.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CustomerServiceImpl implements CustomerService {
+
+    private CustomerRepository customerRepository;
+    @Autowired
+    public CustomerServiceImpl(CustomerRepository customerRepository) {
+        this.customerRepository = customerRepository;
+    }
+
+    @Override
+    public Optional<Customer> getCustomerById(Long id) {
+        return customerRepository.findById(id);
+    }
+
+    @Override
+    public List<Customer> getAllCustomers() {
+        return customerRepository.findAll();
+    }
+
+    @Override
+    public Customer updateCustomerById(Long id, Customer customerDetails) {
+        Customer customer = customerRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Customer not found with id " + id));
+
```

Comment:
What if you will have object with more than 50 fields? Use more generic solution